Number_types/include/CGAL/FPU.h

```diff
@@ -503,12 +518,18 @@ FPU_get_and_set_cw (FPU_CW_t cw)
 // A class whose constructor sets the FPU mode to +inf, saves a backup of it,
 // and whose destructor resets it back to the saved state.
 
+#ifdef CGAL_ALWAYS_ROUND_TO_NEAREST
+#define CGAL_FE_PROTECTED CGAL_FE_TONEAREST
+#else
+#define CGAL_FE_PROTECTED CGAL_FE_UPWARD
+#endif
+
 template <bool Protected = true> struct Protect_FPU_rounding;
 
 template <>
 struct Protect_FPU_rounding<true>
 {
-  Protect_FPU_rounding(FPU_CW_t r = CGAL_FE_UPWARD)
+  Protect_FPU_rounding(FPU_CW_t r = CGAL_FE_PROTECTED)
```

Comment:
The protector should not do anything in the new mode. The partial specialization for the protected mode could be hidden behind the macro.

Reply:
I think we should consider testing that we are using the expected rounding mode, and error otherwise.